Update Slicing Concepts exercise (part of issue #460)

betsybookwyrm opened the pull request:
Updates the Slicing Concepts exercise of the Variables lesson to:
- use concrete examples
- use the standard term 'stop' for the end of a range in Python
- add an example of using two negative indices

---

Addresses the Slicing Concepts part of issue #460 , but not the other ideas in there. Follows the model given by @mforneris, and also adds the double negative indices example suggested by @brianmatthewthomas .

---

I haven't managed to get `make lesson-check` work for me (I assume that's the best way for me to test my changes don't break the page formatting locally before submitting?) so this is unchecked, sorry.

Also, sorry about the whitespace changes, I'm having trouble committing the file without making those changes for some reason.

If either of these are problems, let me know and I'll put more time into fixing them!

alee commented on _episodes/02-variables.md:
```diff
@@ -334,20 +334,31 @@ print(ewr_422_yY, 'is', flabadab, 'years old')
-> > 4. `thing[:]` returns all of `thing`
-> > 5. `thing[number:some-negative-number]` returns a slice from `number` to `some-negative-number` values from the end of `thing`
-> > 6. If a part of the slice is out of range, the operation does not fail. `atom_name[0:15]` gives the same result as `atom_name[0:]`.
+> > 1.  `species_name[3:8]` returns the substring `'acia b'`
```
minor typo, this should be `[2:8]`

betsybookwyrm replied:
Thanks for catching that!

alee commented on _episodes/02-variables.md:
```diff
@@ -334,20 +334,31 @@ print(ewr_422_yY, 'is', flabadab, 'years old')
+> > 4.  `species_name[:]` returns the entire string `'Acacia buxifolia'`
+> > 5.  `species_name[11:-3]` returns the substring `'fo'`, from the 11th position to the third last position
+> > 6.  `species_name[-5:-3]` also returns the substring `'fo'`, from the fifth last position to the third last
+> > 7.  If a part of the slice is out of range, the operation does not fail. `species_name[0:20]` gives the same result as `species_name[0:]`
```
maybe add an `and species_name[:20]` to this line?

alee reviewed (changes requested):
Thank you for the PR! I like the concrete example, and think this is a solid improvement to the challenge. 

I'd never tried `make lesson-check` until now :sweat_smile:  but it does look like it failing due to a missing ruby dependency that wasn't installed on my local system. 

If you've got Docker installed `make docker-serve` should work to render the lesson and bring up a local webserver that lets you see if the formatting still looks OK at localhost:4000

Thanks again!

betsybookwyrm commented:
I've incorporated the typo fix and the suggestion above :) Also rebased on gh-pages branch.

Also, thanks @alee for the tip about `make docker-serve`, I got that working, it's very nice!
